asset/asset.go

```diff
@@ -232,6 +232,71 @@ func DecodeGenesis(r io.Reader) (Genesis, error) {
 	return gen, err
 }
 
+// Specifier is a type that can be used to specify an asset by its ID, its asset
+// group public key, or both.
+type Specifier struct {
```

Comment:
Could also do:
```
type Specifier {
      fn.Either[ID, btcec.PublicKey]
}
```

Then we inherit all the methods of the either type.

Reply:
I'm not sure if `fn.Either` will do here, because sometimes the current code expects both fields to be specified.